WooCommerce/Classes/Authentication/PostSiteCredentialLoginChecker.swift

```diff
@@ -56,8 +56,9 @@ private extension PostSiteCredentialLoginChecker {
                 analytics.track(event: .ApplicationPassword.applicationPasswordGenerationFailed(scenario: .generation, error: error))
                 switch error {
                 case ApplicationPasswordUseCaseError.applicationPasswordsDisabled:
-                    // show application password disabled error
-                    let errorUI = applicationPasswordDisabledUI(for: siteURL)
+                    // show application password disabled error, and catch the last view controller that was showing before the Application Password flow.
+                    let previousViewController = navigationController.viewControllers.dropLast().last
```

Comment:
What is the exact controller that you are navigating to here? Please note that the `PostSiteCredentialLoginChecker` can be triggered from two places: (1) after site credential login (if application password is not required), and (2) after application password login. 

For (1), I'd suggest sending `nil` as the previous view controller to the `PostSiteCredentialLoginChecker`, so that the error page would simply pop itself upon navigating back.

For (2), I'd suggest sending the previous view controller from the application password entry point in `AuthenticationManager`

Reply:
Thank you for your suggestions @itsmeichigo, they are gold and I wasn't 100% aware of the flow. Can you check if the changes make sense now, please? 6bdb62db7e1266caa2dd63dc920b502d2ad83f4b